Meetups/21/2024-11-14-minutes.md

```diff
@@ -0,0 +1,170 @@
+
+Chris: I've also played a little bit by creating different games.
+
+Chris: The whole presentation that I'm showing can be controlled as a WoT Thing
+
+Chris: I have a great number of features already implemented but my free time is running out.
+
+> Christian shows features
+
+Chris: I still have a longer list of features that I want to implement.
+
+Chris: If you want to help me in this effort you can ping me and send your things! Or just reach me online or at WoT Week.
+
+Chris: That was all!
+
+Ege: thank you!
+
+### Q&A
+
+Ege: how did you bridge demo work? how did you control the demo?
+
+Chris: I used a second screen where I controlled the web thing.
+
+Ege: nice to see a lot of different stuff, I think your presentation demonstrate that we can use Web of Things to model services.
+
+Henk: Impressive, a little bit intimidating. How did you generated all the TDs for this examples? Have you automated it?
+
+Chris: all written by hand. But TDs are lazy loaded and can be generated with variables. They are not that complicated.
+
+Ege: About the light demo with channels, you showed that you come up with better interface. How did you do that? Can we generalize this approach?
+
+Chris: is based in the URL but it is not really generalizable.
+
+Cris: can you use Thing Model for this purpose?
+
+Chris: yes but I still have to deal with the mapping at the server side.
+
+Ege: It is not same affordance that is always forwarding to the same channel. Like single request splitting in two other actions in the backend.
+
+Chris: not really, but I have actions that does this (write all 1s or 0s).
+
+Henk: You've done a lot of work with properties, but what about events? have you worked with them?
+
+Chris: I haven't worked a lot with them, but I worked more with actions.
+
+Ege: we will put more guidelines for deciding when to used event or observable properties.
+
+Ege: You started this year and doing in your free time. How easy was to work with WoT, any friction?
+
+Chris: I don't know how much I've spent on this, but way over 100 hours. In terms of difficulty, handle single properties was easy but handling multiple properties was harder.
+
+Chris: Multiple events and query all actions is not clear. For example, the payloads of meta operations and how they should be implemented.
+
+Ege: Is everything open source?
+
+Chris: Still have to decide the future on this project, the code is not really ready for production and I have decide the licensing. Working hard to getting this in useful state for others.
+
+Ege: Any tips for UI generation?
+
+Chris: Point out which libraries there are. It was fun for me, since I am interesting. Objects and Arrays are tricky, especially if they have more than two layers.
+
+Ege: Can you show Godot Engine part a bit more?
+
+Chris: (shows the game) It has a manual fetch and observe buttons.
```

Comment:
```suggestion
Chris: Still have to decide the future of this project. The code is not really ready for production, and I have to decide the licensing. Working hard to get this into a useful state for others.

Ege: Any tips for UI generation?

Chris: Point out which libraries there are. It was fun for me, since I am interested. Objects and Arrays are tricky, especially if they have more than two layers.

Ege: Can you show Godot Engine part a bit more?

Chris: (shows the game) It has manual fetch and observe buttons.
```